src/lib/viz/dataview/mode/DataViewLocal.ts

```diff
@@ -14,12 +14,14 @@ export class DataViewLocal extends DataViewMode {
     this.bindEvents();
   }
 
-  public getSourceData(columns: string[] = [], options: { filterId?: string } = {}) {
+  public async getSourceData(columns: string[] = [], options: { filterId?: string } = {}) {
     if (!columns.includes(this.column)) {
       columns.push(this.column);
     }
 
     if (this.useViewport) {
+      await (this.dataSource as Layer).addSourceField(this.column);
```

Comment:
Why is this conditioned just to useViewport?

Reply:
Yeah, this is a good point and probably an smell.

In the Dataviews, `useViewport` is always `true` except in the case of GeoJSON + `remote`. So, it is going to enter here when:
- MVT + local (it is relevant here)
- GeoJSON + local (it does not matter)
- GeoJSON + remote filtered (it does not matter)

Maybe it could be a good idea to move out of the if in order to avoid all of this. But also, refactoring `useViewport`

Reply:
I think this should be tackled in the second PR that @manmorjim will do (internal refactor), after the current WIP in the public API change. So I will merge this one, and discuss in the context of the new 'mode parameters'